html/infrastructure/safe-passing-of-structured-data/structured-cloning-error-stack-optional.window.js

```diff
@@ -0,0 +1,51 @@
+stackTest(() => {
+  return new Error('some message');
+}, 'page-created Error');
+
+stackTest(() => {
+  return new DOMException('InvalidStateError', 'some message');
+}, 'page-created DOMException');
+
+stackTest(() => {
+  try {
+    Object.defineProperty();
+  } catch (e) {
+    return e;
+  }
+}, 'JS-engine-created TypeError');
+
+stackTest(() => {
+  try {
+    HTMLParagraphElement.prototype.align;
+  } catch (e) {
+    return e;
+  }
+}, 'web API-created TypeError');
+
+stackTest(() => {
+  try {
+    document.createElement('');
+  } catch (e) {
+    return e;
+  }
+}, 'web API-created DOMException');
+
+
+function stackTest(errorFactory, description) {
+  async_test(t => {
+    const error = errorFactory();
+    const originalStack = error.stack;
```

Comment:
Should we add a pointer here to the relevant JavaScript issue for defining this?